arelle/plugin/validate/NL/rules/fr_nl.py

```diff
@@ -21,6 +23,55 @@
 
 _: TypeGetText
 
+BOM_MATCHES = frozenset([
+    codecs.BOM,
+    codecs.BOM32_BE,
+    codecs.BOM32_LE,
+    codecs.BOM64_BE,
+    codecs.BOM64_LE,
+    codecs.BOM_BE,
+    codecs.BOM_LE,
+    codecs.BOM_UTF16,
+    codecs.BOM_UTF16_BE,
+    codecs.BOM_UTF16_LE,
+    codecs.BOM_UTF32,
+    codecs.BOM_UTF32_BE,
+    codecs.BOM_UTF32_LE,
+    codecs.BOM_UTF8,
+])
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NT16,
+        DISCLOSURE_SYSTEM_NT17,
+        DISCLOSURE_SYSTEM_NT18,
+    ],
+)
+def rule_fr_nl_1_01(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation] | None:
+    """
+    FR-NL-1.01: A BOM character MUST NOT be used.
+    """
+    modelXbrl = val.modelXbrl
+    encoding = 'utf-8'
+    for doc in modelXbrl.urlDocs.values():
+        if doc.type == ModelDocument.Type.INSTANCE:
+            with modelXbrl.fileSource.file(doc.filepath, encoding=encoding)[0] as file:
+                firstByte = file.read(1).encode(encoding)
```

Comment:
Looks like I was overly concerned about the unexpected UTF16/32 BOMs, but after testing, Arelle would fail long before this validation is reached. Empty files as well (I don't know if they'd qualify as INSTANCE type docs anyway).

> the underlying openXmlFileStream and XmlUtil.encoding do much more than we need

My approach uses the same method of loading the file, but I don't know if I want to rely on the `encoding` returned as an indication of whether or not a BOM exists, since [XmlUtil.encoding](https://github.com/Arelle/Arelle/blob/2e79b956186c633bf9fc728ad1a536da803a4a04/arelle/XmlUtil.py#L147-L171) can feasibly return values other than `utf-8-sig` when a BOM exists.